modules/programs/alacritty.nix

```diff
@@ -49,14 +315,23 @@ in {
     (mkIf cfg.enable {
       home.packages = [ cfg.package ];
 
-      xdg.configFile."alacritty/alacritty.yml" = mkIf (cfg.settings != { }) {
+      xdg.configFile."alacritty/alacritty.yml" = let
+        settings = if (cfg.CSIuSupport && (cfg.settings ? "key_bindings")) then
+          cfg.settings // {
+            settings.key_bindings =
+              mkMerge [ cfg.settings.key_bindings CSIuKeyBindings ];
```

Comment:
mkMerge is only used in the context of resolving the value of a configuration option.
This won't work as you expect it will since the merging has already been done at this stage.

I think you could add the keybindings like this in the `mkIf cfg.enable` section:

``` nix
programs.alacritty.settings.key_bindings = mkIf cfg.CSIuSupport CSIuKeyBindings;
```

Because `key_bindings` is a list, the merging function for lists will be used.
You also wouldn't need to check if `cfg.settings.key_bindings` exists.

Reply:
> You also wouldn't need to check if cfg.settings.key_bindings exists.

That's the key, thank you.